File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    var answer = [0, 0];
    
    let s_length = 0;
    while (s.length > 1) {
        s_length = s.length;
        
        s = s.split("0").join("");
        
        answer[0]++; // cnt
        answer[1] += s_length - s.length; // zero (원래 크기 - 0을 뺀 크기 = 0 크기)
      
        s = s.length.toString(2); // 2진수 문자열로
    }
    
    return answer;
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    
    int zero=0, cnt=0;
    while(s.length() > 1){
        string tmp = "";
        for(int i=0; i<s.length(); i++){
            if(s[i] == '0')
                zero++;
            else
                tmp += s[i];
        }
        
        int tmp_size = tmp.length();
        vector<int> binary;
        while(tmp_size > 0){
            binary.push_back(tmp_size%2);
            tmp_size /= 2;
        }
        
        s.clear();
        for(int i=binary.size()-1; i>=0; i--)
            s += to_string(binary[i]);
        
        cnt++;
    }
    
    answer.push_back(cnt);
    answer.push_back(zero);
    return answer;
}